fish-rust/src/common.rs

```diff
@@ -115,3 +115,75 @@ pub fn valid_func_name(name: &wstr) -> bool {
 pub const fn assert_send<T: Send>() {}
 
 pub const fn assert_sync<T: Sync>() {}
+
+/// Asserts that a slice is alphabetically sorted by a [`&wstr`] `name` field.
+///
+/// Mainly useful for static asserts/const eval.
+///
+/// # Panics
+///
+/// This function panics if the given slice is unsorted.
+///
+/// # Examples
+///
+/// ```rust
+/// const COLORS: &[(&wstr, u32)] = &[
+///     // must be in alphabetical order
+///     (L!("blue"), 0x0000ff),
+///     (L!("green"), 0x00ff00),
+///     (L!("red"), 0xff0000),
+/// ];
+///
+/// assert_sorted_by_name!(COLORS, 0);
+/// ```
+macro_rules! assert_sorted_by_name {
```

Comment:
Whoa, I did not expect this to be possible in Rust. We may need to disable this check on older versions, but very cool!